### ISSUE TITLE
[lit-html] Duplicate attributes on a node can cause exception in template rendering

### ISSUE THREAD

**Issue by cuberoot** [Polymer/lit-html#2241]
# Description

If you use the same attribute twice on an element and it has a bound value you can cause the template to throw an exception.

I stepped through the code [here](https://github.com/lit/lit/blob/main/packages/lit-html/src/lit-html.ts#L739). The list of attributes gets out of sync and later nodes start trying to expand attributes from previous nodes, causing an error.

# Steps to Reproduce

1. Write this code

```javascript
// Import lit-html functions
import { html, render } from 'lit';

const disabled = false;
const length = 8;
const placeholder = 'Name';

// Define a template function
const myTemplate = (name) =>
  html`
    <h1>Hello ${name}!</h1>
    <input ?disabled=${disabled} length=${length} ?disabled=${disabled} placeholder=${placeholder}>
    <input ?disabled=${disabled}>
  `;

// Render the template with some data
render(myTemplate('lit-html'), document.body);
```

2. See this output...

    Error: Cannot read properties of null (reading 'split')

*Note* the template combines the two `?disabled` attributes into one and this causes the counter to be off by one. The code then tries to look for the `placeholder` attribute on the second `input`, which doesn't have that attribute.

## Live Reproduction Link

https://stackblitz.com/edit/lit-html-ffrddt?file=src/index.js

## Expected Results

I would expect a console warning about a duplicate attribute, but not an exception

## Actual Results

    Error: Cannot read properties of null (reading 'split')

## Browsers Affected

- [x] Chrome
- [x] Firefox
- [x] Edge
- [x] Safari 11
- [x] Safari 10
- [x] IE 11

**Comment by cuberoot** [comment]
Yes, it is definitely a call to get lit-analyzer running on our code. I was given this problem in our code to track down because someone else couldn't figure out what was wrong. On a node with lots of attributes, it can be really hard to see the duplicate. And, the error is not helpful 😄

**Comment by vicb** [comment]
Same here, see [this playground](https://lit.dev/playground/#project=W3sibmFtZSI6InNpbXBsZS1ncmVldGluZy50cyIsImNvbnRlbnQiOiJpbXBvcnQge2h0bWwsIGNzcywgTGl0RWxlbWVudH0gZnJvbSAnbGl0JztcbmltcG9ydCB7Y3VzdG9tRWxlbWVudCwgcHJvcGVydHl9IGZyb20gJ2xpdC9kZWNvcmF0b3JzLmpzJztcblxuQGN1c3RvbUVsZW1lbnQoJ3NpbXBsZS1ncmVldGluZycpXG5leHBvcnQgY2xhc3MgU2ltcGxlR3JlZXRpbmcgZXh0ZW5kcyBMaXRFbGVtZW50IHtcbiAgc3RhdGljIHN0eWxlcyA9IGNzc2BwIHsgY29sb3I6IGJsdWUgfWA7XG5cbiAgQHByb3BlcnR5KClcbiAgbmFtZSA9ICdTb21lYm9keSc7XG5cbiAgcmVuZGVyKCkge1xuICAgIHJldHVybiBodG1sYDxwIFxuICAgICAgICBAbW91c2VlbnRlcj0keygpID0-IGNvbnNvbGUubG9nKCdlbnRlcicpfSBcbiAgICAgICAgQG1vdXNlZW50ZXI9JHsoKSA9PiBjb25zb2xlLmxvZygnbGVhdmUnKX0gPkhlbGxvLCAke3RoaXMubmFtZX0hPC9wPmA7XG4gIH1cbn1cbiJ9LHsibmFtZSI6ImluZGV4Lmh0bWwiLCJjb250ZW50IjoiPCFET0NUWVBFIGh0bWw-XG48aGVhZD5cbiAgPHNjcmlwdCB0eXBlPVwibW9kdWxlXCIgc3JjPVwiLi9zaW1wbGUtZ3JlZXRpbmcuanNcIj48L3NjcmlwdD5cbjwvaGVhZD5cbjxib2R5PlxuICA8c2ltcGxlLWdyZWV0aW5nIG5hbWU9XCJXb3JsZFwiPjwvc2ltcGxlLWdyZWV0aW5nPlxuPC9ib2R5PlxuIn0seyJuYW1lIjoicGFja2FnZS5qc29uIiwiY29udGVudCI6IntcbiAgXCJkZXBlbmRlbmNpZXNcIjoge1xuICAgIFwibGl0XCI6IFwiXjIuMC4wXCIsXG4gICAgXCJAbGl0L3JlYWN0aXZlLWVsZW1lbnRcIjogXCJeMS4wLjBcIixcbiAgICBcImxpdC1lbGVtZW50XCI6IFwiXjMuMC4wXCIsXG4gICAgXCJsaXQtaHRtbFwiOiBcIl4yLjAuMFwiXG4gIH1cbn0iLCJoaWRkZW4iOnRydWV9XQ)

I had two `mouseenter` instead of `enter` and `leave` and it took me a while to debug :(

@justinfagnani would be a nice one to fix / warn about in dev
